### PR TITLE
AZP: enable csclng checks

### PR DESCRIPTION
**Pull request by amaslenn**
## What
Enable `csclng` checks in AZP.

## Why ?
Cover RHEL static checks in UCX CI.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12093/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>UNKNOWN</b> on 16 workers (click for details)</summary>

**Note:** the logs will be deleted after 04-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/29b57c259327afc53ae9a43ea91af2765bf5201c/_main) | :question: ABORTED
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/d0ce39476f35d404271f3c440e3460edb2eb2c4f/hpc-test-althca_W1) | :question: ABORTED
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/21aa016d61396da526bc9c7b6c4270f7fd4fd3c5/hpc-test-althca_W2) | :question: ABORTED
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/f096c2c306af7a8b32ce7f9272c1b17ca22c2ec3/hpc-test-althca_W3) | :question: ABORTED
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/a4a939d5b771a79009bb1a1e2567444537707977/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/c3d7b33e44c1b036ff0980616ff0879a755e6756/hpc-arm-cavium-jenkins_W0) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/c6eb6759cbcc00eb5b1361d346e62e09d60c0050/hpc-arm-cavium-jenkins_W1) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/28e17e936940dbf39a5a288335caed472d85b5ac/hpc-arm-cavium-jenkins_W2) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/c57dd2f71c33bf727425814137562af5b14eb492/hpc-arm-cavium-jenkins_W3) | :question: UNKNOWN
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/6c1ff739fc9047460bba7121429819c725cfe1fb/hpc-test-node-gpu_W0) | :question: UNKNOWN
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/013a296dd94847e27de6c349172a20ced36000d0/hpc-test-node-gpu_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/7a2d81141d3dbefec43b4a138e55dd7d4a2d931c/hpc-test-node-gpu_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/5774bed3d127621a4176351230623caff702daee/hpc-test-node-gpu_W3) | :question: UNKNOWN
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/cf918bcfacc688966194b256aa0b4f0b011021ec/hpc-test-node-new_W3) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/38af26fbbc2f5c9d87fccfc876ce4785a2caa1b3/r-vmb-ppc-jenkins_W0) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/57c6edbd48ef6c19012e0718c833e697/raw/c9155c62afee6a81fb8cd76b4c6e7180a70c20db/r-vmb-ppc-jenkins_W2) | :question: UNKNOWN

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12096/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>ABORTED</b> on 1 workers (click for details)</summary>

**Note:** the logs will be deleted after 04-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/3921727d6322ae6da10c003ac600a928/raw/719f0e12c4ecdc9b4807c114bee4529fda55dc8f/_main) | :question: ABORTED

File: buildlib/fedora.Dockerfile [hunk 1]
```diff
@@ -20,3 +21,7 @@ RUN dnf install -y \
     rdma-core-devel \
     rpm-build \
     && dnf clean dbcache packages
+
+RUN cd `csclng --print-path-to-wrap` && \
+    ln -s cc clang && \
```
[the remark this answers is not in the document]
Comment (amaslenn) [reply buildlib/fedora.Dockerfile]: Most likely I don't need it, checking the best approach now.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12098/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>UNKNOWN</b> on 12 workers (click for details)</summary>

**Note:** the logs will be deleted after 04-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/8ecf67e66389a0ab05ec2c9c3bb5903fc9c9d130/_main) | :question: ABORTED
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/451252975859be65627fe8da4f6ff5f79c4d962f/hpc-test-althca_W0) | :question: ABORTED
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/9594167e07141ed9cb2cc7fdc6e7c75163d58182/hpc-arm-cavium-jenkins_W0) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/33d4ab116f46bb3cdbfff4c0f77729750d4597b5/hpc-arm-cavium-jenkins_W2) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/93c6b02bef4e5a4f96c09707e8959c4479746751/hpc-arm-cavium-jenkins_W3) | :question: UNKNOWN
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/9f01178633c998c0a4dbc60e5eadadbc91568685/hpc-test-althca_W1) | :question: UNKNOWN
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/aebfedea645922eefe4dffb26e214a53692cb551/hpc-test-althca_W2) | :question: UNKNOWN
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/650decde7c7c66b4ec450eab4c836f5d55a9c578/hpc-test-althca_W3) | :question: UNKNOWN
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/0e8984388fe01baf083ce300aa30e0f7fad5b108/hpc-test-node-gpu_W0) | :question: UNKNOWN
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/0fd4f8a35eca47d9caf9eefa519b5da751aa4ff2/hpc-test-node-gpu_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/33972b2d43d16a0706c1465d090f449cec79999d/hpc-test-node-gpu_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/23d74d73ac608fac99fa73f5064cf1bf/raw/ebe3b68076ddd885d59ba6be073208a5f5db1602/hpc-test-node-gpu_W3) | :question: UNKNOWN

**Comment by dmitrygx** [comment]
@amaslenn thank you for discovering a way to reproduce new clang warning!
I've cherry-picked the commit from this PR to test fixes for Clang warnings in #4115 PR
When #4115 is ready, I remove the commit

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12101/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 04-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/12d2319a8129a9e85276f786ba02b83fb19e0063/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/e36f1f119cabc41fd4223134c8d5ff2e449d3ea8/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/69f06eb21a2969286b2101959ad78c021c7870aa/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/1c682bb1c4f5370d76c17d86d6742e4a65a36259/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/acabb442f4bfc40f5562d8fc7223f74db80183b3/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/d5241355091149a8424a472d9ff6cdafe228d954/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/6ff16d28e1389a7be8d156390c12db57a4cfe82c/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/0e12a9c16bc9874944d61b3534e261423c2a02ad/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/b7b19779341ca34b2c29fff6ed6cf22166dfa2b8/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/86d7a9452fccd94e14a598a5ca6a59ab95d84749/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/005b3e010d95e53f420bb7768d1ab367ac96ad9f/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/2d7ae52b1fb8d1858637cf95a819cf5443344bd4/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/e42c9398ea13907358ee99a5252a6ec5ba388b48/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/2c35259cf1ea6a8fa84e3d26a5c0184c985d1e39/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/0625642dc6a380a2326777db92771f5fc8da564d/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/84da4805fe22a44df062b1dd6cd9ba9c4410a3f6/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/132d58a15b25bd1b354e6f81f1c4f617a1c1d8aa/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/57e4c2bdff16db2d5d70c08954ecbf92235ae394/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/e9afecdf3f605d225425a76c85646106f36f2bb2/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/71c7a70630fee4f014a509c5289ac9156677314d/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/be3de7617426742f2470000d29ed56d25acae2bb/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/f6e56b44c10e085bf36efc474aa5483f1b59ca59/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/4a308ba7afa4376f1d76bad40a1e20b511c3bbd0/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/3a07b4f91eee90dff9f7291c1d37c6b19b1988b8/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/b63179bdbe1aae3448ceb1d73fd4131b87351a0d/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/916b18c5de8ed4a5b2a9336d7e9dd7daf45d657d/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/25b386331387149caeb07ae884ea7f6fb4ebca62/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/cf445f9b7bd8c3b86821c65b3caa8c408aeb3c74/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/32a19fa03a482b514c3bc5ccd8fb953e/raw/4869203ee878604a4bec4cb1da568fa7b76d76b8/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
@jgunthorpe this PR is will not be merged until the errors are fixed.

File: buildlib/azure-pipelines.yml [hunk 2]
```diff
@@ -19,61 +19,52 @@ resources:
...
+            displayName: Configure
 
           - bash: |
               set -eE
 
-              cd build-gcc
-              cppcheck --version
+              cd build
+
+              export PATH="`csclng --print-path-to-wrap`:`cscppc --print-path-to-wrap`:`cswrap --print-path-to-wrap`:$PATH"
```
[the remark this answers is not in the document]
Comment (amaslenn) [reply buildlib/azure-pipelines.yml]: You mean this:
```sh
$ csclng
Usage:
    export PATH="`csclng --print-path-to-wrap`:$PATH"

    csclng is a compiler wrapper that runs clang in background.  Create a
    symbolic link to csclng named as your compiler (gcc, g++, ...) and put it
    to your $PATH.  csclng --help prints this text to standard error output.
```

`cs*` tools are wrappers around real compilers. Can do
```
$ which gcc clang cppcheck
/usr/lib64/csclng/gcc
/usr/lib64/cswrap/clang
/usr/lib64/cswrap/cppcheck
```
to show its location if you think it is useful.

**Comment by yosefe** [comment]
@dmitrygx there is still one failure:
```
[command]/usr/bin/bash --noprofile --norc /__w/_temp/69701c33-be25-485a-83bd-90e01ce95793.sh
static checkers found errors:
Error: CLANG_WARNING:
/src/ucs/config/parser.c:1005:12: warning: Potential leak of memory pointed to by 'key'
#    *added = 1;
#           ^
##[error]static checkers found errors
##[section]Finishing: cstools reports
```
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=1133&view=logs&jobId=c3b1c751-b3c6-5682-a32e-d1fa8fde7511

**Comment by dmitrygx** [comment]
> @dmitrygx there is still one failure:

@yosefe done
ok to merge?

**Pull request merged**